src/Symfony/Component/Validator/Tests/Constraints/ValidTest.php

```diff
@@ -32,4 +34,34 @@ public function testGroupsAreNullByDefault()
 
         $this->assertNull($constraint->groups);
     }
+
+    /**
+     * @requires PHP 8
+     */
+    public function testAttributes()
+    {
+        $metadata = new ClassMetaData(ValidDummy::class);
+        $loader = new AnnotationLoader();
+        self::assertTrue($loader->loadClassMetadata($metadata));
+
+        [$bConstraint] = $metadata->properties['b']->getConstraints();
+        self::assertFalse($bConstraint->traverse);
+        self::assertSame(['traverse_group'], $bConstraint->groups);
+
+        [$cConstraint] = $metadata->properties['c']->getConstraints();
+        self::assertSame(['my_group'], $cConstraint->groups);
+        self::assertSame('some attached data', $cConstraint->payload);
+    }
+}
+
+class ValidDummy
+{
+    #[Valid]
+    private $a;
+
+    #[Valid(groups: ['traverse_group'], traverse: false)] // Needs a group to work at all for this test
```

Comment:
Why is the group needed?

Reply:
See https://github.com/symfony/symfony/blob/2d44f229ff9cb169c4c90f2c05e60069350eb119/src/Symfony/Component/Validator/Mapping/GenericMetadata.php#L145, if `$constraint->groups === null` the constraint itself is not added but the traversalstrategy is changed. Since we want to check if the argument properly propagates to the constraint (and not the traversalstrategy or other functional behavior) I added a group to demonstrate.